grafana/grafana
# InfluxDB: Fix retention policy handling for InfluxDB 3.0 engine by bringing back the hardcoded default

**What is this feature?**

When we try to fetch the retention policies on [InfluxDB's new engine (3.0)](https://www.influxdata.com/blog/introducing-influxdb-3-0/) we got `feature not implemented 'SHOW RETENTION POLICIES'` error. If we write the retention policy manually it works. So in order to support the new engine properly and also support the old versions this PR is bringing back the old functionality (partially) we removed here https://github.com/grafana/grafana/pull/63820
So even if no retention policy is specified Grafana and InfluxDB will make sure to use the default retention policy. 

Current implementation was lacking to fetch the retention policies therefore it causes a bad user experience. This PR fill fix the issues and improve the UX. 

**Why do we need this feature?**

To be able to support retention policies in all versions of InfluxDB.

**Who is this feature for?**

All InfluxDB users

**Special notes for your reviewer:**

### How to test
- Test `#1`
  - Create a new InfluxDB Cloud account https://cloud2.influxdata.com/signup
  - On the main branch create a new InfluxDB datasource with InfluxQL query language
  - Go explore the page and observe the retention policy couldn't be fetched
- Test `#2`
  - Create a dashboard 
  - Create a template variable for InfluxDB datasource 
  - Create a panel in the dashboard using a template variable 
  - Pick the existing InfluxDB datasource and make the query and save the dashboard.
  - Change the datasource to the new InfluxDB datasource you created in test #1 and save the dashboard
  - Refresh and observe there will be no data
- On this branch, those will be gone and everything will work as expected. 


Please check that:
